### PR TITLE
Removes sleepers

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

Removed sleepers from all maps as well as from ruins and one random room.

## Why It's Good For The Game

Our medbay will be semi-funcional now and will offer more gameplay than toss person into sleeper press a few buttons and that's it.

## Changelog
:cl:
del: Removed sleepers from all maps, few ruins and random rooms.
/:cl:

### CODE REVIEW

**Comment by mcmeiler** [comment]
Didn't know we're /tg/ now.

**Comment by Archanial** [comment]
> 
> 
> Didn't know we're /tg/ now.

cope

**Comment by mcmeiler** [comment]
> cope

What a mature response I would expect from an admin + maintainer.

Make them work on chemistry bags instead. Fill the bag with bottles of reagents, put it in a sleeper, enjoy *limited* sleeper usage until the chems run out. Stock parts increase chem efficiency.

**Comment by TeomanTheGreat** [comment]
> ## Why It's Good For The Game
> 
> 

You know how it says "Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding" when you are making the pr? Im not seeing a legit reason for a total removal when you can just rework or balance it.

**Comment by Archanial** [comment]
I made a rework pr and it was rejected. Why is it good for the game should be obvious and I'm fairly sure it will be obvious for maintainers.

**Comment by ToasterBan** [comment]
could you at least replace the sleepers with stasis beds? maybe even have them be a late game research tech?

**Comment by mcmeiler** [comment]
> I made a rework pr and it was rejected.

* #4056

That's not a sleeper rework.

**Comment by Archanial** [comment]
> 
> 
> could you at least replace the sleepers with stasis beds? maybe even have them be a late game research tech?

They are replaced where needed, like in meta's medbay for example.

**Comment by FriendlyContractor** [comment]
Finally. Sleepers completly ruin any sort of doctor-patient interaction and dumb down treatment to having random assistant shove you into machine and heal almost all type of damage. Not to mention making chemistry largely obsolete when upgraded

**Comment by AnCopper** [comment]
Finally.

**Comment by mcmeiler** [comment]
> Not to mention making chemistry largely obsolete when upgraded

Implying chemistry does their job in the first place.


> having random assistant shove you into machine and heal almost all type of damage

If I saw a greyshit doing medical work without approval I'd kick his ass and throw him out.

**Comment by AnCopper** [comment]
> > Not to mention making chemistry largely obsolete when upgraded
> 
> Implying chemistry does their job in the first place.

They don't do their job because the sleeper literally makes them not have to do it, see the problem now?

**Comment by mcmeiler** [comment]
> because the sleeper literally makes them not have to do it, see the problem now?

They don't do their job because either

* They're dead because they overdosed themselves
* They're dead because they blew themselves up
* They're SSD because they're antag rolling
* They're missing because they're running around everywhere, shooting meth at people

Chemistry as it is right now is nothing but a shitter role.

**Comment by mcmeiler** [comment]
There's a million different useful things chemistry can make already, but chooses not to because they're busy doing their own shit that nobody asked for.

**Comment by AnCopper** [comment]
> > because the sleeper literally makes them not have to do it, see the problem now?
> 
> They don't do their job because either
> 
> * They're dead because they overdosed themselves
> * They're dead because they blew themselves up
> * They're SSD because they're antag rolling
> * They're missing because they're running around everywhere, shooting meth at people
> 
> Chemistry as it is right now is nothing but a shitter role.

They're overdosing themselves on drugs because they don't have anything else to do because the sleeper already does everything for them regarding healing chems, you're telling us more problems caused by the sleeper.

**Comment by mcmeiler** [comment]
> They're overdosing themselves on drugs because they don't have anything else to do because the sleeper already does everything for them regarding healing chems, you're telling us more problems caused by the sleeper.

If medbay as at all robust, they'd ask for superior healing chems, a better cryo mix, etc. Chemists right now simply don't have the mental capacity to do those things, since all they've ever done as chemist is make meth, shoot up meth and die of a meth OD.

Unless chemist mentality was to *suddenly change* (which it won't lmfao), this doesn't make sense to begin with. 

Also fix #5499 before doing this at least.

**Comment by AnCopper** [comment]
> If medbay as at all robust, they'd ask for superior healing chems, a better cryo mix, etc. Chemists right now simply don't have the mental capacity to do those things, since all they've ever done as chemist is make meth, shoot up meth and die of a meth OD.
> 
> Unless chemist mentality was to _suddenly change_ (which it won't lmfao), this doesn't make sense to begin with.

They're dumb because they don't have to make any chems because the sleeper does it for them, worked on literally every other server in making Chemists more useful after the sleeper was removed, you're just listing even more problems caused by the sleeper.

Why would anyone bother with making faster healing chems when the sleeper has an infinite supply of normal healing chems?

**Comment by mcmeiler** [comment]
> > 
> 
> They're dumb because they don't have to make any chems because the sleeper does it for them, worked on literally every other server in making Chemists more useful after the sleeper was removed, you're just listing even more problems caused by the sleeper.

I don't think you've read all comments here. I've already suggested a rework for sleepers that includes chemists in the process. But I wouldn't expect you to since all you read was "sleeper removal" and went "yes".

I might just PR it out of spite.

**Comment by AnCopper** [comment]
> > > 
> > 
> > 
> > They're dumb because they don't have to make any chems because the sleeper does it for them, worked on literally every other server in making Chemists more useful after the sleeper was removed, you're just listing even more problems caused by the sleeper.
> 
> I don't think you've read all comments here. I've already suggested a rework for sleepers that includes chemists in the process. But I wouldn't expect you to since all you read was "sleeper removal" and went "yes".
> 
> I might just PR it out of spite.

The reworks that got denied? Yeah.

**Comment by francinum** [comment]
No.

If you're doing this you're leaving the code intact and removing them from the maps and disabling their ability to be printed.

You are *NOT* going to make nasty custom behavior that breaks consistency.

File: code/game/machinery/Sleeper.dm [hunk 1]
```diff
@@ -28,12 +28,17 @@
...
 /obj/machinery/sleeper/Initialize()
 	. = ..()
 	occupant_typecache = GLOB.typecache_living
 	update_icon()
 	reset_chem_buttons()
 
+/obj/machinery/sleeper/Destroy()
+	. = ..()
+	qdel(circuit)
+
```
Comment (Archanial) [inline code/game/machinery/Sleeper.dm]:
```suggestion
```

**Comment by Crossedfall** [comment]
🔑  terrible way to address the problem

**Comment by francinum** [comment]
I have some bad takes on game design but god damn.

This isn't solving anything, it's just tearing out a piece of equipment because you didn't like the rework offered.

**Comment by Archanial** [comment]
I offered that rework

**Pull request closed without merging**